code/defines/obj/weapon.dm

```diff
@@ -390,50 +390,128 @@
 			else
 				user.show_message("<span class='rose'>There are already two weights on this [src]!</span>")
 
-/obj/item/weapon/legcuffs/beartrap
+/datum/locking_category/beartrap
+		flags = LOCKED_CAN_LIE_AND_STAND
+
+/obj/item/weapon/beartrap
 	name = "bear trap"
 	throw_speed = 2
 	throw_range = 1
-	layer = OPEN_DOOR_LAYER
+	siemens_coefficient = 1
+	plane = OBJ_PLANE
+	layer = BELOW_CLOSED_DOOR_LAYER
+	icon = 'icons/obj/items.dmi'
 	icon_state = "beartrap0"
 	desc = "A trap used to catch bears and other legged creatures."
+	flags = FPRINT
+	origin_tech = Tc_MATERIALS + "=1"
 	starting_materials = list(MAT_IRON = 50000)
 	w_type = RECYK_METAL
+	w_class = W_CLASS_LARGE
+	anchored = FALSE
+	var/breakouttime = 60
 	var/armed = 0
+	var/trapped = 0
+	var/datum/organ/external/trappedorgan = "LIMB_LEFT_LEG"
+	var/mob/living/trappeduser
 	var/obj/item/weapon/grenade/iedcasing/IED = null
 	var/image/ied_overlay
 
-/obj/item/weapon/legcuffs/beartrap/New()
+/obj/item/weapon/beartrap/New()
 	..()
 	ied_overlay = image('icons/obj/items.dmi')
 	ied_overlay.icon_state = "beartrap_ied"
 
-/obj/item/weapon/legcuffs/beartrap/armed
+/obj/item/weapon/beartrap/armed
 	armed = 1
+	anchored = TRUE
```

Comment:
This has some weird properties, on one hand I can see why you've done this (to prevent cheesy dragging around buffed beartraps) but at the same time it also means they can't be pushed around by ZAS etc. Maybe they should just snap shut if moved?

This is not a required change but something to think about.

Reply:
Is there a good way to check if an object is moving? It shouldn't be too hard to just set it off if it moves by a tile.

Reply:
We have an on_moved event which I believe is good for all atom/movable, you may ask in the coder discord